core/src/main/kotlin/ch/ergon/dope/resolvable/expression/unaliased/aggregator/ArrayAggExpression.kt

```diff
@@ -0,0 +1,22 @@
+package ch.ergon.dope.resolvable.expression.unaliased.aggregator
+
+import ch.ergon.dope.DopeQuery
+import ch.ergon.dope.resolvable.expression.unaliased.aggregator.AggregateQuantifier.ALL
+import ch.ergon.dope.resolvable.expression.unaliased.type.Field
+import ch.ergon.dope.resolvable.operator.FunctionOperator
+import ch.ergon.dope.validtype.ValidType
+
+class ArrayAggExpression(
+    private val field: Field<out ValidType>,
+    private val quantifier: AggregateQuantifier,
+) : AggregateExpression, FunctionOperator {
+    override fun toDopeQuery(): DopeQuery {
```

Comment:
There is a lot of duplication in these classes. I think it's better if you define the AggregateExpression as a class and move all this code there. Then you can write here something like in all the subclasses:
```
class ArrayAggExpression(
    field: Field<out ValidType>,
    quantifier: AggregateQuantifier,
) :  AggregateExpression(field, quantifier, "ARRAY_AGG")
```
where you only have to pass the aggregator-name ARRAY_AGG

Reply:
done